### ISSUE TITLE
How to increase NPairsLoss batch size. 

### ISSUE THREAD

**Issue by marued** [KevinMusgrave/pytorch-metric-learning#197]
This is more of a question:

When using  NPairsLoss, it's creating unique pairs before running the Cross Entropy Loss (just like in the paper). Is there a way to **increase the batch size** by adding more examples for each class? Or will the loss function only keep one example of each class? Are all the negative examples kept? 

For example, can I feed a representation tensor of size N * M * V, where N is the batch size, M the number of different examples for a single class and V the embedding size?

**Comment by KevinMusgrave** [comment]
Reshape your embeddings to be size (N*M, V), and the labels to be shape (N*M, V), and then use [NTXentLoss](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#ntxentloss).

**Comment by marued** [comment]
Thanks for the quick response. I also had a feeling I should use NTXentLoss instead. Is it because NTXentLoss better supports bigger batches or is there another theoretical reason I'm missing?

**Comment by KevinMusgrave** [comment]
No, it's just a different implementation. When there's only 1 positive pair per anchor, you can do a simple matrix operation, but when there are multiple positive pairs it gets a bit more complicated.

**Comment by KevinMusgrave** [comment]
I should maybe add an explanation to the documentation.

**Comment by KevinMusgrave** [comment]
I added this to the docs: "If your batch has more than 2 samples per label, then you should use NTXentLoss."
